zbus_macros/src/iface.rs

```diff
@@ -1293,16 +1293,43 @@ fn introspect_add_output_args(
         if let Type::Tuple(t) = ty {
             if let Some(arg_names) = arg_names {
                 if t.elems.len() != arg_names.len() {
-                    // Turn into error
-                    panic!("Number of out arg names different from out args specified")
+                    return Err(Error::new_spanned(
+                        ty,
+                        format!(
+                            "Number of out arg names ({}) different from out args specified ({})",
+                            arg_names.len(),
+                            t.elems.len()
+                        ),
+                    ));
```

Comment:
Kudos for fixing the existing code too but no good deed goes unpunished :-) so you gotta split this. ;)